### PR TITLE
Feature/additional params

### PR DESCRIPTION
**Pull request by Yamaguchi**
This PR allows the wallet to add extra parameters when the wallet sends token.
These parameters are added in request headers or request body in JSON-RPC over HTTP

### CODE REVIEW

File: types/wallet.d.ts [hunk 1]
```diff
@@ -7,17 +7,24 @@ import { KeyStore } from './key_store';
 import { Rpc } from './rpc';
 import { TransferParams } from './token';
 import { Utxo } from './utxo';
+export interface BroadcastOptions {
+    headers?: {
+        [key: string]: string;
+    };
+    params?: any[];
+}
+export declare type TransferOptions = BroadcastOptions;
```
Comment (azuchi) [inline types/wallet.d.ts:16]: Is this additional parameter only called broadcast?
For example, the name BroadcastOptions may not be appropriate if you also need to call it when communicating with listUnspent.

File: types/wallet.d.ts [hunk 2]
```diff
@@ -29,10 +36,10 @@ export declare class BaseWallet implements Wallet {
     importExtendedPrivateKey(xpriv: string): Promise<void>;
     importWif(wif: string): Promise<void>;
     update(): Promise<void>;
-    broadcast(tx: tapyrus.Transaction): Promise<string>;
+    broadcast(tx: tapyrus.Transaction, options?: BroadcastOptions): Promise<string>;
```
Comment (azuchi) [inline types/wallet.d.ts:39]: If it is a communication parameter that is not directly related to the operation of the wallet, it is better to be able to get it internally as a configuration value, not as an argument.

Comment (Yamaguchi) [reply types/wallet.d.ts:39]: This is intended that backend application needs additional parameter such as "Receipt ID" or "Gift Code" that differ between each transaction. So i think it is better to add parameters as an argument than configuration

File: types/wallet.d.ts (hunk 1 above)
Comment (Yamaguchi) [reply types/wallet.d.ts:16]: Yes. we need it only when broadcasting (as for now).

**Review by azuchi — APPROVED** [review]
